libs/sdk-flutter/lib/breez_sdk.dart

```diff
@@ -14,82 +14,56 @@ class BreezSDK {
 
   /* Streams */
 
-  /// Listen to paid Invoice events
   final StreamController<InvoicePaidDetails> _invoicePaidStream = StreamController.broadcast();
 
+  /// Listen to paid Invoice events
   Stream<InvoicePaidDetails> get invoicePaidStream => _invoicePaidStream.stream;
 
-  /// Listen to payment results
   final StreamController<Payment> _paymentResultStream = StreamController.broadcast();
 
+  /// Listen to payment results
   Stream<Payment> get paymentResultStream => _paymentResultStream.stream;
 
-  /// Initializes SDK events & log streams
+  /* SDK Streams */
+
+  StreamSubscription<BreezEvent>? _breezEventsSubscription;
+  StreamSubscription<LogEntry>? _breezLogSubscription;
+
+  Stream<BreezEvent>? _breezEventsStream;
+  Stream<LogEntry>? _breezLogStream;
+
+  /// Initializes SDK events & log streams.
+  ///
+  /// Call once on your Dart entrypoint file, e.g.; `lib/main.dart`.
   void initialize() {
     _initializeEventsStream();
     _initializeLogStream();
   }
 
-  /// Listen to BreezEvent's(new block, invoice paid, synced)
   void _initializeEventsStream() {
-    _lnToolkit.breezEventsStream().listen((event) async {
-      if (event is BreezEvent_InvoicePaid) {
-        _invoicePaidStream.add(event.details);
-        await fetchNodeData();
-      }
-      if (event is BreezEvent_Synced) {
-        await fetchNodeData();
-      }
-      if (event is BreezEvent_PaymentSucceed) {
-        _paymentResultStream.add(event.details);
-      }
-      if (event is BreezEvent_PaymentFailed) {
-        _paymentResultStream.addError(PaymentException(event.details));
-      }
-      if (event is BreezEvent_BackupSucceeded) {
-        _backupStreamController.add(event);
-      }
-      if (event is BreezEvent_BackupStarted) {
-        _backupStreamController.add(event);
-      }
-      if (event is BreezEvent_BackupFailed) {
-        _backupStreamController.addError(BackupException(event.details));
-      }
-      if (event is BreezEvent_SwapUpdated) {
-        _swapEventsStreamController.add(event);
-      }
-    });
+    _breezEventsStream = _lnToolkit.breezEventsStream().asBroadcastStream();
   }
 
-  /// Listen to node logs
   void _initializeLogStream() {
     if (defaultTargetPlatform == TargetPlatform.android) {
-      const EventChannel('breez_sdk_node_logs')
+      _breezLogStream = const EventChannel('breez_sdk_node_logs')
           .receiveBroadcastStream()
-          .map((log) => LogEntry(line: log["line"], level: log["level"]))
-          .listen(
-            (log) => _logStreamController.add(log),
-            onError: (e) => _logStreamController.addError(e),
-          );
+          .map((log) => LogEntry(line: log["line"], level: log["level"]));
     } else {
-      _lnToolkit.breezLogStream().listen((logEntry) {
-        _logStreamController.add(logEntry);
-      }, onError: (e) {
-        _logStreamController.addError(e);
-      });
+      _breezLogStream = _lnToolkit.breezLogStream().asBroadcastStream();
```

Comment:
@erdemyerebasmaz  I think I understand now what you are trying to do here. As far as I understand now this line actually calls the rust side logic in bindings.rs. Is that right?

Reply:
Yes, that's true and we ensure it's called once, create a broadcast stream reference of it that can be subscribed/unsubscribed from via their StreamSubscription's.